galley/pkg/config/analysis/analyzers/auth/mtls.go

```diff
@@ -82,30 +80,17 @@ func (s *MTLSAnalyzer) Analyze(c analysis.Context) {
 	// TODO Reuse pilot logic as a library rather than reproducing its logic
 	// here.
 
+	mc := util.MeshConfig(c)
+
 	// If autoMTLS is turned on, bail out early as the logic used below does not
 	// reason about its usage.
-	autoMtlsEnabled := false
-	rootNamespace := "istio-system"
-
-	// Only one MeshConfig should exist in practice - we use ForEach to avoid
-	// specifying where to look for the MeshConfig (which allows the Context
-	// object to provide it to us).
-	c.ForEach(metadata.IstioMeshV1Alpha1MeshConfig, func(r *resource.Entry) bool {
-		mc := r.Item.(*meshconfig.MeshConfig)
-		if mc.GetEnableAutoMtls() != nil && mc.GetEnableAutoMtls().Value {
-			autoMtlsEnabled = true
-		}
-
-		if mc.GetRootNamespace() != "" {
-			rootNamespace = mc.GetRootNamespace()
-		}
-		return true
-	})
-
-	if autoMtlsEnabled {
+	if mc.GetEnableAutoMtls().GetValue() {
```

Comment:
Was this purely a refactor or does it change behavior? Some of the expected test output changed below.

Reply:
1. A test was previously broken, and didn't fail even after it should have. Thus the `s/istio-policy/istio-telemetry`.
2. There's a behavior change where for the mtls analyzer we look at the `istio=annotation` to tell us what resources are control plane and should be skipped, instead of relying on the namespace. This has the dual benefit of avoiding needing to know the `istio-system` namespace, and also lets us detect errors for non-control-plane resources that are bundled in the namespace (like Grafana). 
3. There's some minor test cleanup to add `{}` to be consistent with what is valid when you `kubectl apply` it.